datadog_checks_base/datadog_checks/base/utils/concurrency/limiter.py

```diff
@@ -0,0 +1,87 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import threading
+
+
+class ConditionLimiter(object):
+    """
+    This class is used to limit the number of concurrent satisfied conditions. This is intended for use in
+    scenarios where it is critical that no conditions are satisfied after a certain limit has been reached.
+    Therefore, ephemeral false negatives are acceptable in order to maintain this guarantee without blocking.
```

Comment:
Not a blocker, but would it make sense to highlight that after the limit was reached, no conditions would be evaluated until an earlier slot was explicitly released?